components/size_aware_image.jsx

```diff
@@ -230,6 +349,20 @@ export default class SizeAwareImage extends React.PureComponent {
         );
     }
 
+    copyLinkToAsset = () => {
+        const fileURL = this.props.fileURL;
+        copyToClipboard(fileURL ?? '');
```

Comment:
I think what we need to do here is that once the user clicks on this button, we call the `getFilePublicLink` action (as seen [here](https://github.com/mattermost/mattermost-webapp/blob/master/components/file_preview_modal/file_preview_modal_main_actions/file_preview_modal_main_actions.tsx#L49)) to get the link and then once that's done, we continue on and copy it.

We have to wait until the user clicks on this button to load the link or else we'll end up making a bunch of requests for those links for every file when we likely won't need it most of those times.

Reply:
On it, thanks for describing here so clearly.

Reply:
Hi @hmhealey could you please check https://github.com/mattermost/mattermost-webapp/pull/9705/commits/509f8930e9301c951421da285d98dc82b21a0bba commit? 

TL;DR: Unlike file preview component, the api call couldn't happen in component mount because they'd be unnecessary. 

So the click handler triggers `getFilePublicLink` and [another function](https://github.com/mattermost/mattermost-webapp/pull/9705/commits/509f8930e9301c951421da285d98dc82b21a0bba#diff-83c6734e3d6d99f358f1bbc40bd17e5a478d44eed57d06e7e1af20f4c31aaef0R354) which runs on render function everytime(on every update) to see if `filePublicLink` is updated. 

If yes, it copies to clipboard and resets `filePublicLink` in global state.

Reply:
Yep, having the download use the internal link makes sense to me. We know the user can see the file, so we don't need to worry about permissions with it.

Unfortunately, that way of copying the link after getting the URL doesn't work. We can't have side effects like that in the render method since React requires it to be pure. It should actually be easier to do that in the click handler itself though. Instead of storing anything in state, I think you should be able to make the click handler itself wait to receive the response from the server by doing something like
```tsx
onClick = () => {
    // You can still set some "copying..." state here if we want to show that to the user
    this.props.actions.getPublicLink(fileId).then((data) => {
        copyToClipboard(data); // You should get the URL returned from the server here, but I'm not sure of the exact format
        // TODO Set state to show it was copied
    });
}
```

That should also reduce the number of props that we need to pass into the `SizeAwareImage` that's specific to this use case. I'm not a huge fan of adding all this functionality into that component since it was fairly self-contained and focused just on sizing and placeholders before, so I'd like to limit the amount we're passing into it for this new functionality.

Reply:
I agree, that was unnecessarily complex. I've pushed a commit to fix that.